chef-config/lib/chef-config/config.rb

```diff
@@ -857,7 +857,7 @@ def self.windows_home_path
       # Full URL to the endpoint that will receive our data. If nil, the
       # data collector will not run.
       # Ex: http://my-data-collector.mycompany.com/ingest
-      default :server_url,       nil
+      default(:server_url) { File.join(config_parent.chef_server_url, "/data-collector") }
```

Comment:
Yes, before this patch is applied, you must explicitly configure this in order for the data collector to be enabled (it's in the old behavior of `DataCollector.register_reporter?()` ).